staging/kos/pkg/controllers/ipam/ipam.go

```diff
@@ -598,25 +598,22 @@ func (ctlr *IPAMController) analyzeAndRelease(ns, name string, att *netv1a1.Netw
 		ipl := ownedObj.(*netv1a1.IPLock)
 		consider(ipl)
 	}
-	if att != nil && att.Status.IPv4 != "" {
+	if att != nil && att.Status.IPv4 != "" && lockInStatus.Obj == nil {
 		// Make sure we do not skip this one just because we have not
 		// yet been notified about it.
 		statusIP := gonet.ParseIP(att.Status.IPv4)
 		if statusIP != nil {
-			statusIPU := convert.IPv4ToUint32(statusIP)
 			statusUsed := float64(0)
 			defer func() { ctlr.statusUsedHistogram.Observe(statusUsed) }()
-			if _, found := considered[statusIPU]; !found {
```

Comment:
This check based on `considered` was unneeded and wrong. Unneeded because it's more straightforward to base the check on `lockInStatus.Obj` (the check has been added a few lines above) which we need anyway, hence we can get rid of `considered`.
Most importantly, using `considered` was also incorrect.  See the deleted code a few lines above which populated `considered` and notice how this was done **before** the check on the owner's UID. Assume there is a time slipper Y locking the same address as the one in the status of the attachment X being processed but in a different VNI, and assume Y's lock is in the informer's cache while X's lock is not. The IP in X's status would be in `considered` because of Y's lock, hence no live look up of X's lock would be performed, which means that thrashing of already assigned IP addresses was still possible.
Notice how the solution based on `lockInStatus` is not the only possible one --- we can move the population of `considered` after the check on the owner's UID or we can change `considered` to store the VNI of IPs as well. But IMO using `lockInStatus` is the most intuitive and straightforward choice.

Reply:
agreed.